packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2323,6 +2325,46 @@ export class ContainerRuntime
 		return result;
 	}
 
+	/**
+	 * Returns an aliased data store with the given alias.
+	 * @param alias - The alias for the data store.
+	 * @returns - The data store as a FluidObject if it exists and is aliased. Returns undefined if the data
+	 * store doesn't exists or is not aliased.
+	 */
+	public async getAliasedDataStore(alias: string): Promise<FluidObject | undefined> {
+		await this.dataStores.waitIfPendingAlias(alias);
+		const internalId = this.internalId(alias);
+		let context: FluidDataStoreContext;
+
+		try {
+			context = await this.dataStores.getDataStore(internalId, { wait: false });
+		} catch (error) {
+			// If getting the data store fails with 404 code, the data store doesn't exist. Return undefined.
+			// Otherwise, its an unexpected failure. Throw the error.
+			if ((error as IResponseException)?.code === 404) {
+				return undefined;
+			}
+			throw error;
+		}
+
+		// If the data store is not an alias, return undefined.
+		if (!(await context.isRoot())) {
+			return undefined;
+		}
+
+		const channel = await context.realize();
+		if (channel.entryPoint !== undefined) {
+			return channel.entryPoint.get();
+		}
+
+		// back-compat - If entryPoint is not defined, fall back to requesting "/".
+		const response = await channel.request({ url: "/" });
```

Comment:
i don't think this code will ever run. if they have this method, they will have entrypoints for datastores, even if the the typing suggests it won't exist. You might consider just making the check above an assert, so `assert(channel.entryPointe !== undefined,"blah")`, and then you don't need this code, which we plan to delete soon anyway.

Reply:
this will make returning the handle easier

Reply:
Good suggestion. Changed it.